lib/tools/adb-commands.js

```diff
@@ -1196,56 +1196,27 @@ methods.killProcessesByName = async function killProcessesByName (name) {
  */
 methods.killProcessByPID = async function killProcessByPID (pid) {
   log.debug(`Attempting to kill process ${pid}`);
-  let wasRoot = false;
-  let becameRoot = false;
+  const noProcessFlag = 'No such process';
   try {
-    try {
-      // Check if the process exists and throw an exception otherwise
-      await this.shell(['kill', '-0', pid]);
-    } catch (e) {
-      if (_.includes(e.stderr, 'No such process')) {
-        return e.stderr;
-      }
-      if (!_.includes(e.stderr, 'Operation not permitted')) {
-        throw e;
-      }
-      try {
-        wasRoot = await this.isRoot();
-      } catch (ign) {}
-      if (wasRoot) {
-        throw e;
-      }
-      log.info(`Cannot kill PID ${pid} due to insufficient permissions. Retrying as root`);
-      ({isSuccessful: becameRoot} = await this.root());
-      try {
-        await this.shell(['kill', '-0', pid]);
-      } catch (e1) {
-        if (_.includes(e1.stderr, 'No such process')) {
-          return e1.stderr;
-        }
-        throw e1;
-      }
+    // Check if the process exists and throw an exception otherwise
+    await this.shell(['kill', pid]);
+  } catch (e) {
+    if (_.includes(e.stderr, noProcessFlag)) {
+      return e.stderr;
     }
-    const timeoutMs = 1000;
-    let stdout;
-    try {
-      await waitForCondition(async () => {
-        try {
-          stdout = await this.shell(['kill', pid]);
-          return false;
-        } catch (e) {
-          // kill returns non-zero code if the process is already killed
-          return true;
-        }
-      }, {waitMs: timeoutMs, intervalMs: 300});
-    } catch (err) {
-      log.warn(`Cannot kill process ${pid} in ${timeoutMs} ms. Trying to force kill...`);
-      stdout = await this.shell(['kill', '-9', pid]);
+    if (!_.includes(e.stderr, 'Operation not permitted')) {
+      throw e;
     }
-    return stdout;
-  } finally {
-    if (becameRoot) {
-      await this.unroot();
+    log.info(`Cannot kill PID ${pid} due to insufficient permissions. Retrying as root`);
+    try {
+      await this.shell(['kill', pid], {
+        privileged: true
+      });
+    } catch (e1) {
+      if (_.includes(e1.stderr, noProcessFlag)) {
+        return e1.stderr;
```

Comment:
why do we want to return the error string?

Reply:
because it was already implemented like that. Do you have better ideas on what could be returned? Currently the result is anyway not used anywhere

Reply:
I guess I would expect it to return nothing if the result is not valuable. And since it's an error message, if we do 'return' it I would expect it to be thrown as an error and not returned as a string.

Reply:
changed it to return nothing